app/app.js

```diff
@@ -3,177 +3,49 @@
-            error.code = 400
-            return error // image is not PNG or JPEG, so it cannot be used
-        }
-    }
-}
-
-/**
- * Handles passing the base64 string to all proper methods, which includes checking its file type and writing the
- * file to disk
- * @param {string} base64String - base64 string of file
- * @returns {*} - file path if the argument provided was PNG or JPEG, null otherwise
- */
-function processImage(base64String) {
-    if (base64String !== null && base64String !== undefined && base64String !== '') {
-        const fileType = checkIfBase64IsJpeg(base64String) // checking the file type of the base64
-        // throws an error if file type is not compatible
-        if (fileType instanceof Error) {
-            return fileType
-        } else {
-            const fileName = uuid() + '.jpeg'
-            // call writeJpegToDisk if true or writePngToDisk if false
-            let writeFile = fileType ? writeJpegToDisk(base64String, fileName) : writePngToDisk(base64String, fileName)
-            if (writeFile instanceof Error) {
-                throw writeFile
-            } else {
-                return '../slothbucket/' + fileName // file path
-            }
-        }
-    } else {
-        let error = new Error('Cannot process null, undefined, or empty string arguments')
-        error.code = 400
-        return error
-    }
-}
-
-// HTTP POST for any TensorFlow image recognition with bonus sloth check
-app.post('/sloth-check', (req, res) => {
-    if (req.body.base64 !== null && req.body.base64 !== undefined && req.body.base64 !== '') {
-        let filePath = processImage(req.body.base64) // sending to check file type and save to disk
-        if (filePath instanceof Error) {
-            // if there's an error, that means the image wasn't able to be processed (e.g., it wasn't a JPEG or PNG)
-            res.status(200).send({status: filePath.code, message: filePath.message})
-        } else {
-            // commands to run the TensorFlow script
-            child_process.execFile('python', ['classify_image.py', '--image_file', filePath], {
-                cwd: '../root' // change working directory
-            }, function(error, result) {
-                if (error !== null) {
-                    res.status(500)
-                        .send({status: 500, message: 'Internal error encountered while running TensorFlow script.'})
+// routes ===============================
+app.post('/sloth-check', (request, response) => {
+    const base64 = request.body.base64
+    if (base64) {
+        // first, write the file to disk
+        writeFile.writeFileToDisk(base64).then((fileName) => {
+            // next, run TensorFlow's trained model
+            runTensorFlow.classifyImage(fileName).then((tensorFlowResult) => {
+                // finally, send the result
+                if (tensorFlowResult instanceof Error) {
```

Comment:
This works, but it's not really node convention. See the above comment for how errors are normally handled in Node

Reply:
This comment still applies though